### ISSUE TITLE
Change way relationships between report and observables are defined

### ISSUE THREAD

**Issue by himynamesdave** [signalscorps/file2stix#11]
Remove the `object_refs` from the Report Objects created. Instead use Relationship Object to define these relationships

there are two types of extractions in Obstracts; default and custom.

In the case of default extractions, a Relationship between the extracted Object and Report SDO is created with the `relationship_type` equal to `default-extract`. The `created` and `modified` dates match those in the linked Report Object.

Here is the structure of the SRO for default extractions;

```json
  {
    "type": "relationship",
    "spec_version": "2.1",
    "id": "relationship--<GENERATED BY STIX2 LIBRARY>",
    "created_by_ref": "identity--<USER IDENTITY ID>",
    "created": "<ITEM INGEST DATE>",
    "modified": "<ITEM INGEST DATE>",
    "relationship_type": "default-extract",
    "source_ref": "report--<REPORT OBJECT>",
    "target_ref": "<EXTRACTED STIX OBSERVABLE ID>"
  },
```

Custom extractions have slightly different Relationship Objects created where the `relationship_type` equal to `custom-extract`, as follows;

```json
  {
    "type": "relationship",
    "spec_version": "2.1",
    "id": "relationship--<GENERATED BY STIX2 LIBRARY>",
    "created_by_ref": "identity--<USER IDENTITY ID>",
    "created": "<ITEM INGEST DATE>",
    "modified": "<ITEM INGEST DATE>",
    "relationship_type": "custom-extract",
    "source_ref": "report--<REPORT OBJECT>",
    "target_ref": "<EXTRACTED STIX OBSERVABLE ID>"
  },
```

**Comment by himynamesdave** [comment]
> object_refs is a required property of stix2.Report class in python library. So I am leaving object_refs field as it is and adding new Relationship SROs in the bundle.

OK.

**Comment by himynamesdave** [comment]
`default-extract` seems to work fine. However, odd behaviour with custom.

Try 

```
file2stix --input-file tests/custom_extractions/test_extractions.txt --custom-extraction-file tests/custom_extractions/test_extractions.txt    
```

In the bundle, a lot of relationship objects exist, but these don't exist in the bundle, as shown below.



```
{
    "type": "bundle",
    "id": "bundle--85c60e95-654b-47d6-bf99-b5eb733b87b6",
    "objects": [
        {
            "type": "tool",
            "id": "tool--0c8465c0-d0b4-4670-992e-4eee8d7ff952",
            "created_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
            "created": "2017-05-31T21:33:06.824Z",
            "modified": "2022-04-20T20:02:41.550Z",
            "name": "at",
            "description": "[at](https://attack.mitre.org/software/S0110) is used to schedule tasks on a system to run at a specified date or time.(Citation: TechNet At)(Citation: Linux at)",
            "labels": [
                "tool"
            ],
            "external_references": [
                {
                    "source_name": "mitre-attack",
                    "url": "https://attack.mitre.org/software/S0110",
                    "external_id": "S0110"
                },
                {
                    "source_name": "Linux at",
                    "description": "IEEE/The Open Group. (2017). at(1p) \u2014 Linux manual page. Retrieved February 25, 2022.",
                    "url": "https://man7.org/linux/man-pages/man1/at.1p.html"
                },
                {
                    "source_name": "TechNet At",
                    "description": "Microsoft. (n.d.). At. Retrieved April 28, 2016.",
                    "url": "https://technet.microsoft.com/en-us/library/bb490866.aspx"
                }
            ],
            "object_marking_refs": [
                "marking-definition--fa42a846-8d90-4e51-bc29-71d5b4802168"
            ],
            "x_mitre_aliases": [
                "at",
                "at.exe"
            ],
            "x_mitre_attack_spec_version": "2.1.0",
            "x_mitre_deprecated": false,
            "x_mitre_domains": [
                "enterprise-attack"
            ],
            "x_mitre_modified_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
            "x_mitre_platforms": [
                "Linux",
                "Windows",
                "macOS"
            ],
            "x_mitre_version": "1.2"
        },
        {
            "type": "report",
            "spec_version": "2.1",
            "id": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "created": "2022-08-28T16:43:27.653675Z",
            "modified": "2022-08-28T16:43:27.653675Z",
            "name": "File converted: test_extractions.txt",
            "report_types": [
                "threat_report"
            ],
            "published": "2022-08-28T17:43:27.653648Z",
            "object_refs": [
                "tool--0c8465c0-d0b4-4670-992e-4eee8d7ff952"
            ]
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--8c066e7f-fad0-4557-9061-9aab05b7934f",
            "created": "2022-08-28T16:43:27.654046Z",
            "modified": "2022-08-28T16:43:27.654046Z",
            "relationship_type": "default-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "tool--0c8465c0-d0b4-4670-992e-4eee8d7ff952"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--b238ec19-3ca5-458f-a45a-ecd6f5899ab1",
            "created": "2022-08-28T16:43:27.654244Z",
            "modified": "2022-08-28T16:43:27.654244Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "attack-pattern--c2411d68-1b92-4fe1-83a6-11d4c7fbb091"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--5a7ad312-8329-4ebf-8a19-af99a6e1e4c8",
            "created": "2022-08-28T16:43:27.654428Z",
            "modified": "2022-08-28T16:43:27.654428Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "campaign--f32b6484-76f8-4bfa-91ce-630bf15e9668"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--80b2d1c6-6af7-4f21-887b-adf5e8196e40",
            "created": "2022-08-28T16:43:27.654687Z",
            "modified": "2022-08-28T16:43:27.654687Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "course-of-action--2cd24c9b-defb-4efc-b4a7-ddf33204493e"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--a15daed9-3f48-42de-a42e-8346d23807b9",
            "created": "2022-08-28T16:43:27.654891Z",
            "modified": "2022-08-28T16:43:27.654891Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "infrastructure--ae987545-15f9-406b-a274-e158a529e4c1"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--dfd7d6e3-756d-4f72-aa54-6c2ec8b7831c",
            "created": "2022-08-28T16:43:27.655078Z",
            "modified": "2022-08-28T16:43:27.655078Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "intrusion-set--999b8c31-bfc6-4a1e-a47f-d420903906b2"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--876e9169-044e-4f3a-bf54-0e87b67e5c8c",
            "created": "2022-08-28T16:43:27.655263Z",
            "modified": "2022-08-28T16:43:27.655263Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "malware--9ffb8e1f-8939-4da4-b39a-a46ea22d0b0a"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--15b4ac08-b1f8-459d-b01f-edc1bb35a8e6",
            "created": "2022-08-28T16:43:27.655446Z",
            "modified": "2022-08-28T16:43:27.655446Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "threat-actor--daefc3ad-b916-44a8-a55b-20143bb7d682"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--f8de6520-f4a8-436c-a398-825638b18d89",
            "created": "2022-08-28T16:43:27.655625Z",
            "modified": "2022-08-28T16:43:27.655625Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "tool--2a3e3e3d-23b6-4cb2-bb62-da99c255f0f4"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--cd8751a4-c50e-433a-9511-27bb504731a1",
            "created": "2022-08-28T16:43:27.655799Z",
            "modified": "2022-08-28T16:43:27.655799Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "infrastructure--1bc0aea4-0e37-4f5f-ae8a-87fa86dbad6a"
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--06f92956-6f9b-49cc-b010-f6bf36a58589",
            "created": "2022-08-28T16:43:27.655971Z",
            "modified": "2022-08-28T16:43:27.655971Z",
            "relationship_type": "custom-extract",
            "source_ref": "report--e22e2d08-577e-4cf7-a55d-a3ca5cc21fe2",
            "target_ref": "intrusion-set--79654635-9269-4044-92cd-b7774a71828b"
        }
    ]
}
```
@shanthanu9
